home.asm

```diff
@@ -785,7 +785,7 @@ INCLUDE "home/random.asm"
 INCLUDE "home/joypad.asm"
 INCLUDE "home/palettes.asm"
 
-Func_dd4: ; 0xdd4
+HorrendousMultiplyAbyL: ; 0xdd4
```

Comment:
It isn't even a horrendous way to multiply. I'm sure what he had in mind as the "good" way to multiply is a small loop that keeps decrementing 'a' so that it adds the input value (register 'l') to 'bc' exactly 'a' times. But a multiplication like that takes variable time to execute and grows linearly with the input value for 'a'.

This multiplication routine has a constant execution time that is short and independent of the input values of 'a' and 'l'. This routine keeps doubling the value in 'bc' (sla c \ rl b), and for each set bit in 'a', add the current value in 'bc' to 'hl'. The result in 'hl' is exactly (a * l) but there are no loops that have to get repeated 'a' times. The routine just goes top to bottom one time and it's done.

If he really insists on calling this routine anything other than just "Multiply", it should really be called "FastMultiply".